modules/tide_edit_protection/src/Form/TideEditProtectionForm.php

```diff
@@ -0,0 +1,165 @@
+use Drupal\Core\Entity\EntityTypeBundleInfo;
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Tide edit protection form.
+ */
+class TideEditProtectionForm extends ConfigFormBase {
+
+  /**
+   * Entity Type Manger.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManger;
+
+  /**
+   * Entity Type Bundle.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfo
+   */
+  protected $entityTypeBundle;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'tide_edit_protection_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['tide_edit_protection.form'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManager $entityManager, EntityTypeBundleInfo $bundleInfo) {
+    parent::__construct($config_factory);
+    $this->entityTypeManger = $entityManager;
+    $this->entityTypeBundle = $bundleInfo;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('config.factory'), $container->get('entity_type.manager'), $container->get('entity_type.bundle.info'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('tide_edit_protection.form');
+    $entity_types = $entity_bundles = [];
+    $_entity_types = $this->entityTypeManger->getDefinitions();
+    foreach ($_entity_types as $entity_type) {
+      $entity_types[$entity_type->id()] = $entity_type->getLabel();
+    }
+    $_entity_bundles = $this->entityTypeBundle->getAllBundleInfo();
+    foreach ($_entity_bundles as $entity_type_id => $bundles) {
+      foreach ($bundles as $bundle_id => $bundle) {
+        $entity_bundles[$entity_type_id . '__' . $bundle_id] = $entity_types[$entity_type_id] . ' - ' . $bundle['label'];
+      }
+    }
+
+    $config_entity_types = $config->get('entity_types');
+    if (empty($config_entity_types)) {
+      $config_entity_types = [];
+    }
+    $config_entity_bundles = $config->get('entity_bundles');
+    if (empty($config_entity_bundles)) {
+      $config_entity_bundles = [];
+    }
+
+    $entity_methods = [
+      '' => $this->t('None'),
+      'all' => $this->t('All entity forms'),
+      'entity_type' => $this->t('Forms by entity type'),
+      'entity_bundle' => $this->t('Forms by entity bundle'),
+    ];
+
+    $form['entity_method'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Method'),
+      '#default_value' => $config->get('entity_method'),
+      '#options' => $entity_methods,
+      '#description' => $this->t('Choose which entity forms will be match for unsaved warning when left form page.'),
```

Comment:
"Choose which entity forms will be matched for unsaved changes warning when leaving form page.